### PR TITLE
Scroll element to center of viewport to avoid bottom overlay interference

### PR DESCRIPTION
**Pull request by vever001**
Following https://github.com/Lullabot/MinkSelenium2Driver/pull/10 and https://github.com/Lullabot/MinkSelenium2Driver/pull/13 logic, I noticed radios and select elements also require a scrollIntoView which are currently missing. Without this, we get MoveTargetOutOfBounds exception when attempting to click on select/radios which are out of the viewport.

### CODE REVIEW

**Comment by alexpott** [comment]
@vever001 thanks for the MR. Any chance you could add test coverage for this? Have a look in \Behat\Mink\Tests\Driver\Custom\LargePageClickTest and the mink driver-test-suite dependency.

**Comment by vever001** [comment]
@alexpott 

I'm a bit confused... I tried to reproduce with tests initially but couldn't.
I then tried with the form from my project and found out that size attributes on form elements might be the cause.

The test passes if we comment out the JS that adds the attribute.
But when we uncomment that line (adding size="200" on the first_name text element) then we get
`WebDriver\Exception\ElementClickIntercepted: Element <input name="sex" type="radio"> is not clickable at point (516,811) because another element <html> obscures it`

It looks like scrollIntoView block: 'center' fixes it but I'm not sure what is going on exactly...

File: src/Selenium2Driver.php [hunk 1]
```diff
@@ -953,7 +953,7 @@ public function mouseOver(string $xpath)
     private function scrollElementIntoView(Element $element): void {
         $script = <<<JS
             var e = arguments[0];
-            e.scrollIntoView({ behavior: 'instant', block: 'end', inline: 'nearest' });
+            e.scrollIntoView({ behavior: 'instant', block: 'center', inline: 'nearest' });
```
Comment (alexpott) [inline src/Selenium2Driver.php:956]: Thinking about this change. I'm wondering why it is necessary?

**Comment by alexpott** [comment]
I think maybe in this instance this might be very related to https://github.com/Lullabot/MinkSelenium2Driver/pull/15 - and given the way your project works maybe it is your projects job to do the necessary scrolling. It's really tricky when people overlap elements that are suppose to receive clicks.

File: src/Selenium2Driver.php (hunk 1 above)
Comment (vever001) [reply src/Selenium2Driver.php:956]: I tried the fix from https://github.com/Lullabot/MinkSelenium2Driver/pull/15 but it didn't solve the issue.
I'll spend some more time shortly to try to understand what's going on but I'm not sure the issue is specific to my project, I think it's an edge case (with size attributes?).
If we revert this line, the test fails.

Comment (AlexSkrypnyk) [reply src/Selenium2Driver.php:956]: @vever001 
#15 does not scroll for you. it only prevents the re-scrolling. If you apply #15, you would need to scroll to the element before making an assertion.

Comment (KalleVuorjoki) [reply src/Selenium2Driver.php:956]: We faced similar issue after upgrading Drupal 10.2 to 10.3, where `behat/mink-selenium2-driver` got replaced with this library. The tests started to fail and patching the `e.scrollIntoView()` block argument to `center` fixed the issues of layouts with floating toolbar receiving the press/click events.

**Comment by vever001** [comment]
I just hit this on another project. This happens somewhat intermittently for elements where the browser has to scroll down. I think it's either:
- a bug or edge case from chrome/firefox with scrollIntoView({ block: 'end' }} as using block: center fixes it (but maybe it's unlikely that both chrome and firefox have the same bug)
- or it's not a browser bug and there is the tiniest layout shift happening after the scroll, causing the element to go just what it needs to be off-page for selenium not to be able to click it (we get: "move target out of bounds"). Meaning `block: 'end'` is brittle on this (layout shifts) and using block: 'center' might be safer.

**Comment by vever001** [comment]
I reverted the previous code, and renamed this issue title.
scrolling for radio/select elements doesn't seem necessary as selenium seems too handle it on its own.

But the scroll center does fix things in 2 projects for me as it makes things more resilient when we have bottom overlays and/or layout shifts.
